Fetch-API - Details

KageKrito opened the pull request:
## Overview
This PR adds the functionality to fetch the details of a specific mission by `missionId`

Resolves #111 
#
## Changes
- Added method `getDetailsByMission` to `fetchapi.ts` to fetch details associated with a specific mission
- Refactor `getDetailsByMission` to implement `getFormattedDetails`, `getTotalDuration` and `getTotalSize`
- Added functionality to `DetailView` to use `getFormattedDetails` to fetch the details with `useEffect`
- Implemented `getTotalDuration` and `getTotalSize` to `DetailView` to use it in `InformationView`
- Implemented `getTotalDuration` and `getTotalSize` to `Overview`
- Added jest tests for `getDetailsByMission`,  `getFormattedDetails`, `getTotalDuration` and `getTotalSize` to `fetchapi.test.ts`
- Added documentation for `getDetailsByMission`, `getFormattedDetails`, `getTotalDuration` and `getTotalSize` to [docs/fetchapi](../tree/main/docs/fetchapi)
- Name change from `"get_files"` to `"get_files_by_mission_id"` in restapi/urls.py and restapi/tests.py

## Testing
Can be tested by clicking on a mission, it should show all details associated to a mission stored in the database.
Also it should show the total duration and total size in the information view on the right and in the overview.

DerSimi reviewed (changes requested):
Hi Oliver,
i went through your changes, and I'm almost happy :)

Backend implementation is fine and it works, documentation is also complete.

However, getDetailsMyMission does include the wrong header. This will lead to a permission denied error when using it with api keys, with it, it works:
![image](https://github.com/user-attachments/assets/54be86c6-788a-4b67-a066-8c36b1dc741c).

Another issue is related to units: In models.py you write:
![image](https://github.com/user-attachments/assets/66eced78-2807-4954-8535-240cb56a447e)

But you display the database data unfiltered:
![image](https://github.com/user-attachments/assets/566ca5e7-346d-49bc-b1ed-c6244b04b45e)

Meaning convert sizes from bytes to megabytes and duration from "nanoseconds" (why nanoseconds?) to hh:mm:ss

For robot name, I'm unsure. It should be displayed in the headline, but this field only exists in File. Ig we discuss this on Thursday if a mission always uses the same robot.

KageKrito commented:
Thank you for your review Simon,
I hope I fixed everything that was requested.
The detailview should now display the duration in the format hh:mm:ss and I changed the unit of duration for the model File to seconds.
Also the size should be converted and displayed in the correct unit with the unit written next to it. 
Therefor I also removed the '(MB)' in the top row of the table.
Last I also changed the headers of `getDetailsByMission` to include all the headers defined in `headers`.

DerSimi reviewed (approved):
Hi Oliver,
thanks for your changes, it works as expected.

![image](https://github.com/user-attachments/assets/ebd39ef5-708c-4b80-8243-c9d85e517d46)

However, the stats on the right are still missing, meaning the accumulated size and duration. Should we do this in a separate PR?

![image](https://github.com/user-attachments/assets/c287b8f3-d83f-40a2-96d5-bdd9e54fbcdb)

As well as show these stats in the mission table.

KageKrito commented:
That`s a good point!
I refactored getDetailsByMission and added new functions to fetch the total duration and size.
Now the Information view and the Overview should both show the correct total duration and time

DerSimi reviewed (approved):
LGTM
